### DISCUSSION TITLE
Access hide/show columns data

### DISCUSSION THREAD

**Discussion by tshailesh-vmw** [vmware/clarity#5592] [Q&A]
How to get data/information about hidden columns after user makes choose to make certain columns hidden or choose to show them?
I have implemented hide/show columns using clrDgHideableColumn directive in Clarity version 3 Datagrid:

```
<clr-dg-column *ngFor="let column of results.columns">
    <ng-container *clrDgHideableColumn="{hidden: column.hide}">
        {{column.value | titlecase}}
    </ng-container>
</clr-dg-column>
```

**Comment by gnomeontherun** [comment]
There are a few issues details suggestions on how to do this.

https://github.com/vmware/clarity/issues/2117
https://github.com/vmware/clarity/issues/4227

**Reply by tshailesh-vmw** [reply]
I tried this 
https://github.com/vmware/clarity/issues/2204#issuecomment-383210477
but its not working for me 
https://stackblitz.com/edit/clarity-v3-dyn-columns-issue-tqrcev?file=src%2Fapp%2Fapp.component.ts

getting this same issue https://github.com/vmware/clarity/issues/2204#issuecomment-539592765
